apps/contracts/scripts/mint-lions.ts

```diff
@@ -159,14 +144,12 @@ async function main() {
 
   const sni = SovereignNatureIdentifier.attach(SNI_CONTRACT_ADDRESS);
 
-  axios.defaults.headers.common[
-    'Authorization'
-  ] = `Bearer ${process.env.NFT_STORAGE_API_KEY}`;
-
   const data = fs.readFileSync('./data/lions_data.json');
   const lionsData: Array<LionData> = JSON.parse(data.toString());
 
-  lionsData.forEach((ld) => mintLionData(ld, sni));
+  for (const ld of lionsData) {
+    await mintLionData(ld, sni); //TODO: There is a strange race condition here, so we need to wait for each transaction to be mined before minting the next one
```

Comment:
You need to do manual verification that the tx went through, this is **always** very painful handling, because you need to run a timer. And yes, you have to wait to send the next tx otherwise you can get wrong nonces. Here is a helper for handlers: https://ethereum.stackexchange.com/questions/6002/transaction-status

Reply:
@cerealkill Thx, will check. I already wait for transaction finish via `tx.wait`, seems stable, but will look into your example.

Reply:
I checked, i tried this before, works the same way as tx.wait() method.